android/src/org/coolreader/crengine/Utils.java

```diff
@@ -89,32 +89,17 @@ private static boolean moveFile(File oldPlace, File newPlace, boolean removeOld)
 			Log.i("cr3", "File " + oldPlace.getAbsolutePath() + " does not exist!");
 			return false;
 		}
-		FileOutputStream os = null;
-		FileInputStream is = null;
-		try {
+		try (FileOutputStream os = new FileOutputStream(newPlace);
```

Comment:
@Isira-Seneviratne You have changed the order of function calls.
Before:
```java
if (!newPlace.createNewFile())
	return false; // cannot create file
os = new FileOutputStream(newPlace);
is = new FileInputStream(oldPlace);
```
After:
```java
FileOutputStream os = new FileOutputStream(newPlace);
FileInputStream is = new FileInputStream(oldPlace);
if (!newPlace.createNewFile())
	return false;
```
So, now this function will always return false.
https://developer.android.com/reference/java/io/FileOutputStream#FileOutputStream(java.io.File)
https://developer.android.com/reference/java/io/File#createNewFile()

Reply:
Okay, I'll fix it.